go/libraries/doltcore/sqle/dprocedures/dolt_backup.go

```diff
@@ -15,366 +15,374 @@
-		}
-	case cli.RemoveBackupId, cli.RemoveBackupShortId:
-		err = removeBackup(ctx, dbData, apr)
-		if err != nil {
-			return statusErr, fmt.Errorf("error removing backup: %w", err)
-		}
-	case cli.RestoreBackupId:
-		if err = restoreBackup(ctx, dbData, apr); err != nil {
-			return statusErr, fmt.Errorf("error restoring backup: %w", err)
-		}
-	case cli.SyncBackupUrlId:
-		err = syncBackupViaUrl(ctx, dbData, sess, apr)
-		if err != nil {
-			return statusErr, fmt.Errorf("error syncing backup url: %w", err)
-		}
-	case cli.SyncBackupId:
-		err = syncBackupViaName(ctx, dbData, sess, apr)
-		if err != nil {
-			return statusErr, fmt.Errorf("error syncing backup: %w", err)
+	funcParam := apr.Arg(0)
+	switch funcParam {
+	case DoltBackupParamAdd:
+		err = doltBackupAdd(ctx, dbData, doltSess, apr)
+	case DoltBackupParamRemove, DoltBackupParamRm:
+		if apr.NArg() != 2 {
+			return nil, errDoltBackupUsage(funcParam, []string{"name"}, nil)
 		}
+		err = dbData.Rsw.RemoveBackup(ctx, apr.Arg(1))
+	case DoltBackupParamSync:
+		err = doltBackupSync(ctx, dbData, doltSess, apr)
+	case DoltBackupParamSyncUrl:
+		err = doltBackupSyncUrl(ctx, dbData, doltSess, apr)
+	case DoltBackupParamRestore:
+		err = doltBackupRestore(ctx, dbData, doltSess, apr)
 	default:
-		return statusErr, fmt.Errorf("unrecognized dolt_backup parameter: %s", apr.Arg(0))
+		return nil, fmt.Errorf("unrecognized %s parameter '%s'", DoltBackupProcedureName, funcParam)
 	}
 
-	return statusOk, nil
+	return rowToIter(int64(0)), err
 }
 
-func addBackup(ctx *sql.Context, dbData env.DbData[*sql.Context], apr *argparser.ArgParseResults) error {
+// doltBackupAdd adds a new backup entry with the name and URL specified in |apr|. The URL is normalized to an absolute
+// path. AWS parameters are extracted from command-line flags in |apr| if present, otherwise they are loaded from
+// session variables if the URL scheme matches.
+func doltBackupAdd(ctx *sql.Context, dbData env.DbData[*sql.Context], dsess *dsess.DoltSession, apr *argparser.ArgParseResults) error {
 	if apr.NArg() != 3 {
-		return fmt.Errorf("usage: dolt_backup('add', 'backup_name', 'backup-url')")
+		return errDoltBackupUsage(DoltBackupParamAdd, []string{"name", "url"}, optionalAwsParams)
 	}
 
-	backupName := strings.TrimSpace(apr.Arg(1))
-	backupUrl := apr.Arg(2)
-	cfg := loadConfig(ctx)
-	scheme, absBackupUrl, err := env.GetAbsRemoteUrl(filesys.LocalFS, cfg, backupUrl)
+	backupName := apr.Arg(1)
+	backupUrlScheme, backupUrl, err := newAbsRemoteUrl(dsess, apr.Arg(2))
 	if err != nil {
-		return fmt.Errorf("error: '%s' is not valid, %s", backupUrl, err.Error())
-	} else if scheme == dbfactory.HTTPScheme || scheme == dbfactory.HTTPSScheme {
-		// not sure how to get the dialer so punting on this
-		return fmt.Errorf("sync-url does not support http or https backup locations currently")
+		return err
 	}
 
-	params, err := cli.ProcessBackupArgs(apr, scheme, absBackupUrl)
+	backupParams, err := newParams(apr, backupUrl, backupUrlScheme)
```

Comment:
These are supported by the CLI, if they're provided we'll just have them override the session vars.